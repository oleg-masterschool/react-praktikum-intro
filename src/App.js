function App(props) {
  const { course, cohort, instructor, beginner = false} = props;
  const coursePrefix = beginner ? 'Beginner' : 'Advanced';

  return (
    <div className="App">
      <article>
        <h1>Welcome to {coursePrefix} { course }!</h1>
        <p>This is cohort number { cohort }.</p>
        <p>Taught by instructor { instructor }.</p>
      </article>
    </div>
  );
}

export { App };
