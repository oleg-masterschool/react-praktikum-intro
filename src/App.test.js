import { render, screen } from '@testing-library/react';
import { App } from './App';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });
// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
// Arrange // Action // Assert
// describe('App snapshot', function () {
//   it('should render the app', function () {
//     const tree = renderer
//         .create(<App/>)
//         .toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
