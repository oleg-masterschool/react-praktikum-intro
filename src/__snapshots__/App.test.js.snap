// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should render the app 1`] = `
<div
  className="App"
>
  <article>
    <h1>
      Welcome to 
      Advanced
       
      !
    </h1>
    <p>
      This is cohort number 
      .
    </p>
    <p>
      Taught by instructor 
      .
    </p>
    <p>
      The best course ever.
    </p>
    <h2>
      Here is the list of 
      candidates
    </h2>
    <div>
      Name - Profession
    </div>
    <ul>
      <li>
        <div
          className="student"
          onClick={[Function]}
        >
          Jerry Seinfeld
           - 
          stand-up comedian
           -
           
          pending
        </div>
      </li>
      <li>
        <div
          className="student"
          onClick={[Function]}
        >
          Cosmo Kramer
           - 
          Kramerica Industries
           -
           
          pending
        </div>
      </li>
      <li>
        <div
          className="student"
          onClick={[Function]}
        >
          Elaine Benes
           - 
          writer
           -
           
          pending
        </div>
      </li>
      <li>
        <div
          className="student"
          onClick={[Function]}
        >
          George Constanza
           - 
          realtor
           -
           
          pending
        </div>
      </li>
    </ul>
    <h2>
      Here is the list of 
      enrolled
    </h2>
    <div>
      Name - Profession
    </div>
    <ul />
  </article>
</div>
`;
